astropy/io/votable/tree.py

```diff
@@ -1546,26 +1549,38 @@ def from_table_column(cls, votable, column):
         Restores a `Field` instance from a given
         `astropy.table.Column` instance.
         """
+        Q_column = None
+        if isinstance(column, Quantity):
+            Q_column = column
+            column = column.info
         kwargs = {}
         for key in ['ucd', 'width', 'precision', 'utype', 'xtype']:
```

Comment:
Above this loop define:
```
meta = column.info.meta or {}
```
Then you can safely assume that `meta` is a dict in the code below and you don't need to be testing for `None`.